samza-azure/src/main/java/org/apache/samza/system/azureblob/compression/CompressionFactory.java

```diff
@@ -25,6 +25,16 @@
  * based on the {@link org.apache.samza.system.azureblob.compression.CompressionType}.
  */
 public class CompressionFactory {
+  private static CompressionFactory compressionFactoryInstance = null;
+  private CompressionFactory() {}
+
+  public static CompressionFactory getInstance() {
```

Comment:
If you would like to do lazy initialization for the singleton, then you will need to be careful about concurrent calls to `getInstance`.
This class doesn't need lazy initialization, so you could just initialize directly when declaring the static variable. Then, you won't have to worry about concurrent calls.